libraries/Microsoft.Bot.Builder.Dialogs/DialogResult.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// Result returned to the caller of one of the various stack manipulation methods and used to
+    /// return the result from a final call to `DialogContext.end()` to the bots logic.
+    /// </summary>
+    public class DialogResult
```

Comment:
DialogResult<T>

Reply:
This isn't a great name but I'm not sure of a better name yet.  `ControlResult` might be a better name but I'd want to do a little more refactoring around how its computed.